app/main/session_parser.py

```diff
@@ -763,6 +763,15 @@ def dirty_sessions_since_clean(uid, mtype):
             if (not clean_found and session_type in [ZSessionType.BEER.value, ZSessionType.COFFEE.value, ZSessionType.SOUS_VIDE.value]):
                 post_clean_sessions.append(s)
 
+        elif mtype == MachineType.ZYMATIC:
+            session_name = session_name_from_filename(s)
+
+            if (session_name.upper() in ["CLEAN", "CLEAN V1", "NEW CLEAN BETA V6"]):
```

Comment:
```suggestion
            if (session_name.upper() in ["CLEAN", "CLEANING V1", "NEW CLEAN BETA V6"]):
```

Reply:
Instead I just changed this to reuse the `r.clean` logic contained within `get_zymatic_recipe_list(clean)` function that was already defined in `routes_zymatic_api.py` by refactoring that into the `routes_frontend.py` file.

Reply:
This way if we or a user were to add a new clean utility recipe then we wouldn't need to modify this string comparison again and again.

Reply:
That made a circular dependency between session parsing and recipe parsing (contained within frontend_routes) so instead went back to my original design (which looking at now is really messy, but works 🤷‍♂️). Updating to the names of recipes as you pointed out works well enough 👍.